soh/soh/Extractor/FastCrc32C.c

```diff
@@ -1,29 +1,39 @@
 #include <stdint.h>
 #include <stddef.h>
 
+// Force the compiler to assume we have support for the CRC32 intrinsic. We will check for our selves later.
+// Clang will define both __llvm__ and __GNUC__ but GCC will only define __GNUC__. So we need to check for __llvm__ first.
+#if ((defined(__llvm__) && defined(__x86_64__) || defined(__i386__)))
```

Comment:
I may not be able to follow the entirety of the code, but this mix of && and || doesn't seem to be properly isolated. I find it hard to follow what the groupings are supposed to be. is it `(llvm && x86_64) || i386`, or `llvm && (x86_64 || i386`?

Reply:
Oh, great, I added them out of order XD.